recipe/build.sh

```diff
@@ -17,15 +25,32 @@ fi
 mkdir -p build
 cd build
 
-cmake \
+cmake ${CMAKE_ARGS} \
   -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
   -DCMAKE_BUILD_TYPE=Release \
   -DLLVM_BUILD_LLVM_DYLIB=ON \
   -DLLVM_LINK_LLVM_DYLIB=ON \
   -DLLVM_BUILD_TOOLS=ON \
   -DLLVM_BUILD_UTILS=ON \
-  -DMLIR_ENABLE_BINDINGS_PYTHON=ON \
-  -DPython3_EXECUTABLE="$PYTHON" \
+  -DMLIR_ENABLE_BINDINGS_PYTHON:bool=ON \
+  -DCMAKE_SKIP_RPATH:bool=ON \
+  -DPYTHON_PACKAGES_PATH="${SP_DIR}" \
+  -DPYTHON_EXECUTABLE="${PYTHON}" \
+  -DPYTHON_INCLUDE_DIR="${INC_PYTHON}" \
+  -DPYTHON_LIBRARY="${LIB_PYTHON}" \
+  -DPYTHON3_EXECUTABLE="${PYTHON}" \
+  -DPYTHON3_INCLUDE_DIR="${INC_PYTHON}" \
+  -DPYTHON3_NUMPY_INCLUDE_DIRS="$(python -c 'import numpy;print(numpy.get_include())')" \
```

Comment:
This ends up picking up the numpy from the build environment, and thus the wrong architecture. Copying from the logs (and reinserting the env vars):
```
-DPYTHON3_NUMPY_INCLUDE_DIRS=$BUILD_PREFIX/venv/lib/python3.8/site-packages/numpy/core/include
                              ^^^^^^
                              !!!!!!
```
It looks like a lot of this is copied from https://github.com/conda-forge/opencv-feedstock/blob/main/recipe/build.sh, but that's not a good reference in this case. Normally you shouldn't have to override all that, as cross-python is supposed to take care of it.

Reply:
Done in https://github.com/conda-forge/mlir-python-bindings-feedstock/pull/32/commits/0f43913ef8c470741cfb52d776a4be7485986430.

Reply:
I meant that we should be dropping way more of the `-DPYTHON*` options here.

BTW, while we iterate, you could add `skip: true  # [build_platform == target_platform]` to the global `build:` section in `meta.yaml`; we don't need to keep dragging along the native builds

Reply:
> while we iterate, you could add `skip: true # [build_platform == target_platform]` to the global `build:` section in `meta.yaml`;

Nice idea! However that seems to skip all builds for some reason now.